vortex-array/src/array/sparse/mod.rs

```diff
@@ -20,11 +20,10 @@ impl_encoding!("vortex.sparse", ids::SPARSE, Sparse);
 
 #[derive(Debug, Clone, Serialize, Deserialize)]
 pub struct SparseMetadata {
-    indices_dtype: DType,
```

Comment:
I think you can remove this, we assert this is constant value when constructing it